tests/defaults.spec.ts

```diff
@@ -0,0 +1,180 @@
+		it('should contain an https protocol', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'apiUrl');
+			m.chai.expect(url.parse(setting).protocol).to.equal('https:');
+		});
+	});
+
+	describe('.vpnUrl', () => {
+		it('should be a valid url', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'vpnUrl');
+			m.chai.expect(() => url.parse(setting)).to.not.throw(Error);
+		});
+
+		it('should not contain a protocol', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'vpnUrl');
+			m.chai.expect(url.parse(setting).protocol).to.not.exist;
+		});
+	});
+
+	describe('.registryUrl', () => {
+		it('should be a valid url', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'registryUrl');
+			m.chai.expect(() => url.parse(setting)).to.not.throw(Error);
+		});
+
+		it('should not contain a protocol', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'registryUrl');
+			m.chai.expect(url.parse(setting).protocol).to.not.exist;
+		});
+	});
+
+	describe('.imageMakerUrl', () => {
+		it('should be a valid url', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'imageMakerUrl');
+			m.chai.expect(() => url.parse(setting)).to.not.throw(Error);
+		});
+
+		it('should contain an https protocol', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'imageMakerUrl');
+			m.chai.expect(url.parse(setting).protocol).to.equal('https:');
+		});
+	});
+
+	describe('.deltaUrl', () => {
+		it('should be a valid url', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'deltaUrl');
+			m.chai.expect(() => url.parse(setting)).to.not.throw(Error);
+		});
+
+		it('should contain an https protocol', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'deltaUrl');
+			m.chai.expect(url.parse(setting).protocol).to.equal('https:');
+		});
+	});
+
+	describe('.dashboardUrl', () => {
+		it('should be a valid url', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'dashboardUrl');
+			m.chai.expect(() => url.parse(setting)).to.not.throw(Error);
+		});
+
+		it('should contain an https protocol', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'dashboardUrl');
+			m.chai.expect(url.parse(setting).protocol).to.equal('https:');
+		});
+	});
+
+	describe('.proxyUrl', () => {
+		it('should be a valid url', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'proxyUrl');
+			m.chai.expect(() => url.parse(setting)).to.not.throw(Error);
+		});
+
+		it('should not contain a protocol', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'proxyUrl');
+			m.chai.expect(url.parse(setting).protocol).to.not.exist;
+		});
+	});
+
+	describe('.dataDirectory', () =>
+		it('should be an absolute path', () => {
+			const setting = utils.evaluateSetting<string>(defaults, 'dataDirectory');
+			const isAbsolute = setting === path.resolve(setting);
+			m.chai.expect(isAbsolute).to.be.true;
```

Comment:
Like in the previous test file, this check and a few of the others below could be `expect(setting).to.equal(path.resolve(setting))`